src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSourceUnitTest.java

```diff
@@ -98,6 +102,34 @@ public void testPartitionSizing(){
         Assert.assertEquals(smallPartitions.partitions().size(), 2);
     }
 
+    @Test
+    public void testReadFromFileAndHDFS() throws IOException {
+        final File bam = new File( getToolTestDataDir(), "reads.bam");
+        final File bai = new File( getToolTestDataDir(), "reads.bai");
+        MiniDFSCluster cluster = null;
+        try {
+            cluster = new MiniDFSCluster.Builder(new Configuration()).build();
+            final Path workingDirectory = cluster.getFileSystem().getWorkingDirectory();
+            final Path bamPath = new Path(workingDirectory,"flag_stat_hdfs.bam");
+            final Path baiPath = new Path(workingDirectory, "flag_stat_hdfs.bai");
+            cluster.getFileSystem().copyFromLocalFile(new Path(bam.toURI()), bamPath);
+            cluster.getFileSystem().copyFromLocalFile(new Path(bai.toURI()), baiPath);
```

Comment:
The entire mini cluster working directory is deleted on exit (see https://github.com/apache/hadoop/blob/trunk/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/MiniDFSCluster.java#L1918, note that the `base_dir` is in _build/test/data/dfs_ by default).